### PR TITLE
fix: create constants for `skipReason`

### LINKED ISSUE #5643: Refactor: Create constants for `skipReason`
Let's define a new constants file for `skipReason`. I think right how we have a lot of inconsistency and duplication. Perhaps though we can first convert these into constants and then afterwards consolidate them.

### PR DESCRIPTION
**Pull request by ViralRuparel**
Moves all the skipReasons to a constant file and replaces those strings with constant from skip-reason constant file.

Closes #5643

### CODE REVIEW

**Comment by ViralRuparel** [comment]
@rarkins It's ready for review!

File: lib/constants/skip-reason.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  UNVERSIONED_REFERENCE: 'unversioned-reference',
+  NOT_A_VERSION: 'not-a-version',
+  IGNORED: 'ignored',
+  FILE_DEPENDENCY: 'file-dependency',
+  MULTIPLE_CONSTRAINT_DEP: 'multiple-constraint-dep',
+  LOCAL: 'local',
+  NO_SOURCE: 'no-source',
+  INTERNAL_PACKAGE: 'internal-package',
+  DISABLED: 'disabled',
+  UNSUPPORTED_VALUE: 'unsupported-value',
```
Comment (rarkins) [inline lib/constants/skip-reason.ts]: @ViralRuparel can you sort these alphabetically? Will make it easier to unify them later. @viceice any thoughts also on whether we should prefix them all (e.g. `SKIP_REASON_LOCAL`) or keep as here?

Comment (ViralRuparel) [reply lib/constants/skip-reason.ts]: sure, let me do it.

Comment (ViralRuparel) [reply lib/constants/skip-reason.ts]: I think we don't need SKIP_REASON as a prefix to each one of them as when we import and use it, we already use it as e.g. `skipReason.UNDEFINED` which makes it clear that it is a skipReason constant.

**Review by rarkins — CHANGES_REQUESTED** [review]
Rename `skipReasonConstants` to `skipReasons`

**Comment by ViralRuparel** [comment]
> Rename `skipReasonConstants` to `skipReasons`

DONE.

File: lib/constants/skip-reason.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+export default {
```
Comment (rarkins) [inline lib/constants/skip-reason.ts]: @viceice would this work as an enum?

Comment (viceice) [reply lib/constants/skip-reason.ts]: yes

**Comment by rarkins** [comment]
@ViralRuparel I did some major refactoring of this to use Enum, can you check it now?

**Comment by rarkins** [comment]
@viceice what capitalization do you think we should use these enums?

- UPPER_CASE
- camelCase
- PascalCase
- snake_case
- kebab-case
- Pascal_Snake

This article uses Pascal_Snake for example: https://blog.logrocket.com/writing-readable-code-with-typescript-enums-a84864f340e9/

**Comment by rarkins** [comment]
https://github.com/basarat/typescript-book/blob/master/docs/styleguide/styleguide.md#enum says use PascalCase for both enum name and members

**Comment by rarkins** [comment]
@ViralRuparel can you do the renames? FYI I found that if you use VSCode and right click "Rename symbol" then VSCode will update all references to it from other files.

**Review by rarkins — CHANGES_REQUESTED** [review]
Convert Enum members to PascalCase

**Comment by viceice** [comment]
I've no preference, `PascalCase` is ok for me.

**Comment by ViralRuparel** [comment]
okay

**Comment by ViralRuparel** [comment]
Converted Enum members to PascalCase.

**Review by rarkins — APPROVED** [review]

**Pull request merged**